x/staking/keeper/msg_server.go

```diff
@@ -476,12 +541,16 @@ func (k msgServer) CancelUnbondingDelegation(goCtx context.Context, msg *types.M
 }
 
 func (k msgServer) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
 	if k.authority != msg.Authority {
 		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, msg.Authority)
 	}
 
+	if err := msg.Params.Validate(); err != nil {
```

Comment:
One nit as well here, actually:
Can you remove the check in `SetParams` (https://github.com/cosmos/cosmos-sdk/blob/b17c2d902e372bf0fe1e9fc327591101e133c294/x/staking/keeper/params.go#L54)  and add a comment like this: https://github.com/cosmos/cosmos-sdk/blob/c1cb7256f5667f88cb5566b29535666f5ee59b71/x/gov/keeper/params.go#L10